test/e2e/daemonsets.go

```diff
@@ -0,0 +1,102 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	kubeadmv1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DaemonsetsSpecInput is the input for EnsureDaemonsets.
+type DaemonsetsSpecInput struct {
+	BootstrapClusterProxy framework.ClusterProxy
+	Namespace             *corev1.Namespace
+	ClusterName           string
+}
+
+// EnsureDaemonsets implements a test that verifies expected Daemonset Pods are running.
+func EnsureDaemonsets(ctx context.Context, inputGetter func() DaemonsetsSpecInput) {
```

Comment:
why does this need to take an inputGetter as a func and not simply a `DaemonsetsSpecInput` as param?

Reply:
To answer this question definitively: I implemented this E2E helper func following the pattern of existing helper funcs.

Do either @nojnhuh or @mboersma know why this pattern exists?

Reply:
Summarizing a discussion: we concluded that this pattern is carried over from CAPI E2E; cycles permitting, we can investigate whether or not this is needed in all the places we've copied it in CAPZ E2E.